crates/issue/src/lib.rs

```diff
@@ -123,7 +122,11 @@ pub mod pallet {
     }
 
     #[pallet::hooks]
-    impl<T: Config> Hooks<T::BlockNumber> for Pallet<T> {}
+    impl<T: Config> Hooks<T::BlockNumber> for Pallet<T> {
+        fn on_runtime_upgrade() -> frame_support::weights::Weight {
+            crate::types::v4::migrate_v3_to_v4::<T>()
```

Comment:
When the last migration failed we said that we would do the [try_runtime](https://docs.substrate.io/reference/command-line-tools/try-runtime/) for future migrations. Can you add the required code for the pre_upgrade and post_upgrade hooks?